nav2_navfn_planner/src/main.cpp

```diff
@@ -16,11 +16,14 @@
 
 #include "nav2_navfn_planner/navfn_planner.hpp"
 #include "rclcpp/rclcpp.hpp"
+#include <rclcpp/clock.hpp>
 
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  auto node = std::make_shared<nav2_navfn_planner::NavfnPlanner>();
+  auto clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
+  auto tf_buffer = std::make_shared<tf2_ros::Buffer>(clock);
+  auto node = std::make_shared<nav2_navfn_planner::NavfnPlanner>(tf_buffer);
```

Comment:
are you thinking that eventually nodes might share a tf_buffer?

Reply:
Yes, buffers are a reasonably high overhead, so its optimal as much as possible to give 1 buffer to as many people as possible. This is a hack since the constructor of the buffer in ROS2 right now requires a clock that should normally come from a node, but I can't make a node without a buffer... so this happened. 

This however probably breaks the API for making `components` since the constructor takes something in. The other option is to have the NavFnPlanner make its own buffer, but for future PRs impending having the buffer outside the class has shared benefits with costmap 2D

Reply:
From discussion with @bpwilcox I'll move the buffer inside.